Mathlib/Tactic/DeprecateMe.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Lean.Expr.Basic
+import Mathlib.Tactic.Lemma
+
+/-!
+#  `deprecate to` -- a deprecation tool
+
+Writing
+```lean
+deprecate to id₁ id₂ ... idₙ
+theorem easy : True := .intro
+```
+where `id₁ id₂ ... idₙ` is a sequence of identifiers produces the `Try this` suggestion:
+```lean
+theorem easy : True := .intro
+
+@[deprecated (since := "YYYY-MM-DD")] alias d₁ := id₁
+
+@[deprecated (since := "YYYY-MM-DD")] alias d₂ := id₂
+...
+
+@[deprecated (since := "YYYY-MM-DD")] alias dₙ := idₙ
```

Comment:
```suggestion
@[deprecated (since := "YYYY-MM-DD")] alias id₁ := d₁

@[deprecated (since := "YYYY-MM-DD")] alias id₂ := d₂
...

@[deprecated (since := "YYYY-MM-DD")] alias idₙ := dₙ
```
right? Consider using less obscure names than `d` and `id`, maybe `alias deprₙ := declₙ`?

Reply:
I think that this is correct: the user provides the new names for all the declarations above the declaration-to-be-deprecated.  The workflow should be
* the script replaces the old name by the new name in the original declaration;
* the new declarations get automatically generated with their autogenerated names (more on this later);
* the script retrieves the old autogenerated names and creates the aliases for the (now-non-existing) old autogenerated names to be deprecated in favour of the new ones that the user provided.

The part that I am not too happy about is that you have to provide the new autogenerated names by hand, rather than autogenerating them.  The issue is not actually retrieving those names, but rather matching the new ones with what the old ones were.  This is why you would have to provide the `to_additive` name of a `to_additivized` declaration.  In the specific case of `to_additive` the matching would be easy, as there is only one non-visible declaration, but with `simps` the matching is trickier.

Reply:
Maybe these names make it clearer:
```lean
/--
info: * Pairings:
#[(`new_name_mul, mul_easy_deprecated), (`new_name_add, add_easy_deprecated)]

Try this:
@[to_additive]
theorem new_name_mul : True :=
  .intro

@[deprecated (since := "2024-06-11")]
alias mul_easy_deprecated := new_name_mul

@[deprecated (since := "2024-06-11")]
alias add_easy_deprecated := new_name_add
-/
#guard_msgs in
deprecate to new_name_mul new_name_add
@[to_additive] theorem mul_easy_deprecated : True := .intro
```

Reply:
Anyway, I agree with the suggestion of using meaningful names for the variable and I am going to update them!